src/MartinGeorgiev/Doctrine/DBAL/Types/ValueObject/Ltree.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MartinGeorgiev\Doctrine\DBAL\Types\ValueObject;
+
+/**
+ * @phpstan-consistent-constructor
+ */
+class Ltree implements LtreeInterface
+{
+    /**
+     * @param list<non-empty-string> $pathFromRoot
+     *
+     * @throws \InvalidArgumentException if the pathFromRoot contains empty strings or is not a list
+     */
+    public function __construct(
+        private readonly array $pathFromRoot,
```

Comment:
Perhaps it be cleaner to use expanding arguments here.
```suggestion
        private readonly ...$pathFromRoot,
```

Reply:
I would agree if the node order wasn't important. What you propose is true for something like:

```
public function addChildren(...$children) {}
```

In $pathFromRoot case, the order of the list contents has meaning: `[ 0 => 'root', 1 => 'child', 2 => 'grand-child' ]`.

Reply:
I am not sure I understand your concern. I'm not too bothered if this stays an array or an argument list - just saying that argument unpacking will preserve the order, and that "root -> child -> grand-child" will not somehow become "root -> grand-child -> child".

Reply:
I would prefer it to stay an array, since it might prevent issues with Symfony serializer, which I don't know the behavior of when confronted with argument unpacking.
But I'll follow your lead on this.

Reply:
Ok, let's leave it as per your preference.